selene-lib/src/lints/mixed_table.rs

```diff
@@ -30,7 +30,7 @@ impl Lint for MixedTableLint {
         for mixed_table in visitor.mixed_tables {
             diagnostics.push(Diagnostic::new_complete(
                 "mixed_table",
-                "mixed tables are not allowed".to_owned(),
+                "mixed tables are heavily discouraged due to factors such as: type safety, readability, compatibility, and debugging".to_owned(),
```

Comment:
The message here does not really adequately explain why they should be avoided. Also, "type safety" is not a thing outside of Luau. I don't think it's ever possible to do that in a small message, so I suggest something more openly vague:

```suggestion
                "mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding".to_owned(),
```

Reply:
Aye aye! *Sorry I forgot that selene isn't a linter for just Roblox*